src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -347,17 +430,43 @@ private static Details getDetails(DetailAST ast) {
                     shouldCheckLastRcurly = true;
                     nextToken = getNextToken(ast);
                 }
-                break;
-            case TokenTypes.LITERAL_CATCH:
+            }
+            else if (tokenType == TokenTypes.LITERAL_CATCH) {
                 nextToken = ast.getNextSibling();
                 lcurly = ast.getLastChild();
                 rcurly = lcurly.getLastChild();
                 if (nextToken == null) {
                     shouldCheckLastRcurly = true;
                     nextToken = getNextToken(ast);
                 }
-                break;
-            case TokenTypes.LITERAL_IF:
+
+            }
+            else {
+                shouldCheckLastRcurly = true;
+                nextToken = getNextToken(ast);
+                lcurly = ast.getFirstChild();
+                rcurly = lcurly.getLastChild();
+            }
+            final Details details = new Details();
```

Comment:
Approach 1. seems better. In my humble opinion, approach 2. would make things look more complex than they are.